examples/ip-f107.rs

```diff
@@ -0,0 +1,211 @@
+// A copy of the `ip.rs` example, but for the STM32F107.
+
+#![no_std]
+#![no_main]
+
+extern crate panic_itm;
+
+use cortex_m::asm;
+use cortex_m_rt::{entry, exception};
+use stm32_eth::{
+    hal::flash::FlashExt,
+    hal::gpio::GpioExt,
+    hal::rcc::RccExt,
+    stm32::{interrupt, CorePeripherals, Peripherals, SYST},
+};
+
+use core::cell::RefCell;
+use cortex_m::interrupt::Mutex;
+
+use core::fmt::Write;
+use cortex_m_semihosting::hio;
+
+use fugit::RateExtU32;
+use log::{Level, LevelFilter, Metadata, Record};
+use smoltcp::iface::{InterfaceBuilder, NeighborCache};
+use smoltcp::socket::{TcpSocket, TcpSocketBuffer};
+use smoltcp::time::Instant;
+use smoltcp::wire::{EthernetAddress, IpAddress, IpCidr, Ipv4Address};
+
+use stm32_eth::{EthPins, RingEntry};
+
+static mut LOGGER: HioLogger = HioLogger {};
+
+struct HioLogger {}
+
+impl log::Log for HioLogger {
+    fn enabled(&self, metadata: &Metadata) -> bool {
+        metadata.level() <= Level::Trace
+    }
+
+    fn log(&self, record: &Record) {
+        if self.enabled(record.metadata()) {
+            let mut stdout = hio::hstdout().unwrap();
+            writeln!(stdout, "{} - {}", record.level(), record.args()).unwrap();
+        }
+    }
+    fn flush(&self) {}
+}
+
+const SRC_MAC: [u8; 6] = [0x00, 0x00, 0xDE, 0xAD, 0xBE, 0xEF];
+
+static TIME: Mutex<RefCell<u64>> = Mutex::new(RefCell::new(0));
+static ETH_PENDING: Mutex<RefCell<bool>> = Mutex::new(RefCell::new(false));
+
+#[entry]
+fn main() -> ! {
+    unsafe {
+        log::set_logger(&LOGGER).unwrap();
+    }
+    log::set_max_level(LevelFilter::Info);
+
+    let mut stdout = hio::hstdout().unwrap();
+
+    let p = Peripherals::take().unwrap();
+    let mut cp = CorePeripherals::take().unwrap();
+
+    let mut flash = p.FLASH.constrain();
+
+    let rcc = p.RCC.constrain();
+    // HCLK must be at least 25MHz to use the ethernet peripheral
+    let clocks = rcc
+        .cfgr
+        .sysclk(32.MHz())
+        .hclk(32.MHz())
+        .freeze(&mut flash.acr);
+
+    setup_systick(&mut cp.SYST);
+
+    writeln!(stdout, "Enabling ethernet...").unwrap();
+
+    let mut gpioa = p.GPIOA.split();
+    let mut gpiob = p.GPIOB.split();
+    let mut gpioc = p.GPIOC.split();
+
+    // PLL3CLK goes to MCO (Main Clock Output) (PA8)
+    let _mco = gpioa.pa8.into_alternate_push_pull(&mut gpioa.crh);
```

Comment:
Could you explain why you think the MCO is necessary ? Ping @franeklubi

Reply:
This is not needed, but was part of the example. We can remove it IMO.

Reply:
If it isn't needed then we should remove it, otherwise the users will think it's necessary.